client/my-sites/sharing/connections/account-dialog.jsx

```diff
@@ -1,186 +1,181 @@
-					<h3 className="account-dialog__connected-accounts-heading">{ this.translate( 'Connected' ) }</h3>
+					<h3 className="account-dialog__connected-accounts-heading">{ this.props.translate( 'Connected' ) }</h3>
 					<ul className="account-dialog__accounts">
 						{ this.getAccountElements( connectedAccounts ) }
 					</ul>
 				</div>
 			);
 		}
-	},
+	}
 
-	getDisclaimerText: function() {
+	getDisclaimerText() {
 		if ( 1 === this.props.accounts.length ) {
 			// If a single account is available, show a simple confirmation
 			// prompt to ask the user to confirm their connection.
-			return this.translate( 'Confirm this is the account you would like to authorize. Note that your posts will be automatically shared to this account.', {
-				context: 'Sharing: Publicize connection confirmation'
-			} );
-		} else {
-			// Otherwise, we assume that multiple connections exist for a
-			// single Keyring connection, and the user must choose which
-			// account to connect.
-			return this.translate( 'Select the account you wish to authorize. Note that your posts will be shared to the selected account automatically.', {
+			return this.props.translate(
+				'Confirm this is the account you would like to authorize. Note that your posts will be automatically shared to this account.', {
+					context: 'Sharing: Publicize connection confirmation'
+				} );
+		}
+
+		// Otherwise, we assume that multiple connections exist for a
+		// single Keyring connection, and the user must choose which
+		// account to connect.
+		return this.props.translate(
+			'Select the account you wish to authorize. Note that your posts will be shared to the selected account automatically.', {
 				context: 'Sharing: Publicize connection confirmation'
 			} );
-		}
-	},
+	}
 
-	onClose: function( action ) {
-		var accountToConnect = this.getAccountToConnect();
+	onClose( action ) {
+		const accountToConnect = this.getAccountToConnect();
 
 		if ( 'connect' === action && accountToConnect ) {
 			this.props.onAccountSelected( this.props.service, accountToConnect.keyringConnectionId, accountToConnect.ID );
 		} else {
 			this.props.onAccountSelected();
 		}
-	},
+	}
 
-	render: function() {
-		var classes = classNames( 'account-dialog', {
-			'single-account': 1 === this.props.accounts.length
-		} ), buttons;
+	render() {
+		const classes = classNames( 'account-dialog', {
+				'single-account': 1 === this.props.accounts.length
+			} ),
+			buttons = [
+				{ action: 'cancel', label: this.props.translate( 'Cancel' ) },
+				{ action: 'connect', label: this.props.translate( 'Connect' ), isPrimary: true }
+			];
 
-		buttons = [
-			{ action: 'cancel', label: this.translate( 'Cancel' ) },
-			{ action: 'connect', label: this.translate( 'Connect' ), isPrimary: true }
-		];
+		if ( this.isSelectedAccountConflicting() ) {
+			this.props.warningNotice( this.props.translate( 'The connection marked {{icon/}} will be replaced with your selection.', {
```

Comment:
Am I correct in understanding that this will change the conflict notice from being inline to being a global notice? Does that work well with the dialog being shown? Perhaps also worth a design review.

Ref: 3513-gh-calypso-pre-oss